app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true,
+    length: {maximum: Settings.user.name.length.maximum}
+  validates :email, presence: true,
+    length: {maximum: Settings.user.email.length.maximum},
+    format: {with: Settings.user.email.VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
Dòng này là gì? Từ đâu xuất hiện, có tác dụng gì em?

Reply:
has_secure_password là lời gọi phương thức của Rails, sử dụng bcrypt để băm mật khẩu an toàn.